testing/test_pdb.py

```diff
@@ -1163,3 +1163,28 @@ def runcall(self, *args, **kwds):
     )
     assert result.ret == 0
     result.stdout.fnmatch_lines(["*runcall_called*", "* 1 passed in *"])
+
+
+def test_pdb_eoferror_without_read(testdir):
+    p1 = testdir.makepyfile(
+        """
+
+        def input_without_read(*args, **kwargs):
+            raise EOFError()
+
+
+        def test(monkeypatch):
+            try:
+                import __builtin__
```

Comment:
I suggest we make this obvious that is Python 2 compatibility code, so when we drop Python 2 we also remove the related code. Suggestion:

```python
def test_pdb_eoferror_without_read(testdir):
    builtin_module = '__builtin__' if six.PY2 else 'builtin'
    p1 = testdir.makepyfile(
        """
        def input_without_read(*args, **kwargs):
            raise EOFError()
        def test(monkeypatch):
            import {builtin_module}
            monkeypatch.setattr({builtin_module}, "input", input_without_read)
            __import__('pdb').set_trace()
        """.format(builtin_module=builtin_module)
    )
```

Reply:
Thanks, amended.